woocommerce/woocommerce-gutenberg-products-block
# Improve the data stores documentation

## What

This PR introduces the following changes to improve the data stores documentation:

- Link all README.md files in `assets/js/data/*` to their corresponding files in `docs/third-party-developers/extensibility/data-store/*`.
- List all available data stored in `docs/third-party-developers/extensibility/data-store/README.md`.
- Ensure that all data store docs in `docs/third-party-developers/extensibility/data-store/*` contain the sections `Overview` and `Usage`.
- Create missing data store docs in `docs/third-party-developers/extensibility/data-store/*`.

Fixes #10583
Fixes #10584
Fixes #10585
Fixes #10586

## Why

The current data store documentation was lacking a clear overview and was distributed in various places. 

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Look up the changed files.
2. Verify that all README.md files from `assets/js/data/*` point to the corresponding docs to `docs/third-party-developers/extensibility/data-store/`.
3. Verify that all docs in `docs/third-party-developers/extensibility/data-store/` are accurate.
4. Verify that the docs of all data stored are mentioned in `docs/third-party-developers/extensibility/data-store/README.md`.
5. Verify that the Check Markdown links GitHub Action does not show any broken links.


## WooCommerce Visibility


## Checklist

Required:

Conditional:

## Changelog

> Update: Improve the data stores documentation
